src/sdk/dslink/link.cc

```diff
@@ -324,11 +323,27 @@ ref_<IncomingListCache> DsLink::list(const string_ &path,
   auto merger = _list_mergers[path];
   return merger->list(std::move(callback));
 }
+
 ref_<IncomingInvokeStream> DsLink::invoke(
     IncomingInvokeStreamCallback &&callback,
     ref_<const InvokeRequestMessage> &&message) {
-  return _client->get_session().requester.invoke(std::move(callback),
-                                                 std::move(message));
+  auto stream = _client->get_session().requester.invoke(
+      ([ this, paged_cache = ref_<IncomingPageCache<InvokeResponseMessage>>() ](
```

Comment:
let's std::move capture the callback instead of setting it to the stream
also if this is capture in a async callback, it's better to have a keepref captured too, so it won't be deleted